tfs/resources.py

```diff
@@ -273,5 +353,108 @@ def workitems(self):
         return self.tfs.get_workitems(self.workitem_ids)
 
 
-class GitRepository(TFSObject):
-    pass
+class GitRepository(UnknownTfsObject):
+    def __init__(self, tfs, raw=None):
+        super().__init__(tfs, raw, "git/repositories/{0}", underProject=False)
+
+
+class Project(UnknownTfsObject):
+    def __init__(self, tfs, raw=None):
+        super().__init__(tfs, raw, "projects/{0}", underProject=False)
+
+    @property
+    def teams(self):
+        return self.tfs.teams(projectId=self.id)
+
+
+class Team(UnknownTfsObject):
+    def __init__(self, tfs, raw=None):
+        super().__init__(tfs, raw, "projects/{0}/teams/{1}", underProject=False)
+
+
+class Build(UnknownTfsObject):
+    def __init__(self, tfs, raw=None):
+        super().__init__(tfs, raw, "build/Builds/{0}", underProject=True)
+
+
+class Definition(UnknownTfsObject):
+    def __init__(self, tfs, raw=None):
+        super().__init__(tfs, raw, "build/Definitions/{0}", underProject=True)
+
+
+class Identity(UnknownTfsObject):
+    def __init__(self, tfs, raw=None):
+        super().__init__(tfs, raw, 'Identities/{0}', underProject=False)
+
+#################################################################################
+# Utilities
+#################################################################################
+
+
+def raw2resource(raw, top=None, tfs=None):
+    """ Convert a raw valie into a TFTObject object.
+
+    Recursively walks a dict structure, transforming the properties into attributes
+    on a new ``TfsObject`` object of the appropriate type
+    (if an ``url`` link is present and class found in the class map)
+    or a ``TopLevelWrapper`` object.
+    """
+    if top is None:
+        top = TopLevelWrapper(raw)
+
+    seqs = tuple, list, set, frozenset
+    for i, j in iteritems(raw):
+        if isinstance(j, dict):
+            if isinstance(top, UnknownTfsObject) and i in top.raw_attrs:
+                setattr(top, i, j)
+            elif 'url' in j:
+                resource = class_for_resource(j['url'])(tfs, j)
+                setattr(top, i, resource)
+            else:
+                setattr(
+                    top, i, raw2resource(j, tfs=tfs))
+        elif isinstance(j, seqs):
+            seq_list = []
+            for seq_elem in j:
+                if isinstance(seq_elem, dict):
+                    if 'url' in seq_elem:
+                        resource = class_for_resource(seq_elem['url'])(
+                            tfs, seq_elem)
+                        seq_list.append(resource)
+                    else:
+                        seq_list.append(
+                            raw2resource(seq_elem, tfs=tfs))
+                else:
+                    seq_list.append(seq_elem)
+            setattr(top, i, seq_list)
+        else:
+            setattr(top, i, j)
+    return top
+
+
+resource_class_map = {
+    r'wit/attachments/[^/]+$': Attachment,
```

Comment:
```python
client2 = tfs.TFSAPI("https://tfs.tfs.com/tfs/", project="Collection/Project", user=user, password=password)
wis = client2.run_query('My Queries/AssignedToMe')
```
generate error. `path` is `None` in the function `class_for_resource`
```
Traceback (most recent call last):
 File "E:/GITHUB/dohq-tfs/test.py", line 34, in <module>
    wis = client2.run_query('My Queries/AssignedToMe')
  File "E:\GITHUB\dohq-tfs\tfs\connection.py", line 198, in run_query
    return self._find_resource(TFSQuery, ids=path)
  File "E:\GITHUB\dohq-tfs\tfs\connection.py", line 67, in _find_resource
    resource._find(ids)
  File "E:\GITHUB\dohq-tfs\tfs\resources.py", line 119, in _find
    self._load(uri)
  File "E:\GITHUB\dohq-tfs\tfs\resources.py", line 128, in _load
    self._parse_raw(result)
  File "E:\GITHUB\dohq-tfs\tfs\resources.py", line 323, in _parse_raw
    super()._parse_raw(raw)
  File "E:\GITHUB\dohq-tfs\tfs\resources.py", line 138, in _parse_raw
    raw2resource(raw, self, self.tfs)
  File "E:\GITHUB\dohq-tfs\tfs\resources.py", line 411, in raw2resource
    resource = class_for_resource(j['url'])(tfs, j)
  File "E:\GITHUB\dohq-tfs\tfs\resources.py", line 452, in class_for_resource
    if re.search(resource, path):
  File "C:\Python34\lib\re.py", line 170, in search
    return _compile(pattern, flags).search(string)
TypeError: expected string or buffer
```

Reply:
Can you post your http get json response content for this test?

Reply:
@Boltyk I added test and mock-result for this to develop branch. The above error is reproduced in the feature branch

https://github.com/devopshq/tfs/blob/develop/tests/test_connection.py#L141-L148

Reply:
Sorry for the delayed response :)

Reply:
fixed
https://github.com/devopshq/tfs/pull/65/commits/5d319110bfc7eac7ae3b6a5a99daaa01a792fdb7